src/server/server.cc

```diff
@@ -137,12 +137,17 @@ Status Server::Start() {
   }
 
   if (config_->cluster_enabled) {
+    auto s = cluster_->LoadClusterNodes(config_->NodesFilePath());
```

Comment:
Hi hulk, the reload make kvrocks a bit more trickey, if:
1. a server(A) mount on a disk crashed
2. user add some machines, and update the cluster version
3. A restarted, load the file, but has lower version. Previously, it can recover and load from other machines, but now, user need drop all it's data.

How can we handle this problem?

Reply:
Hi @mapleFU, thanks for your warm review. 

Currently, the cluster topology is expected to maintain out of the Kvrocks node(Like kvrocks-controller) and the external component will re-sync the cluster topology if it's changed. 

So for question 1/2, it's not a problem since the external component will re-sync if it's changed or lost.

For question 3, we can regard this topology as a local cache, the external controller component will re-sync the cluster topology and won't depend on this persistent topology.

Briefly, this PR is only to cache the cluster topo in the local disk, so it will bring the side effect is the user may get stale data if the client cached the topology and the controller failed to update the topology in time. But I think it's better than before.

Reply:
@git-hulk Let's dive into the code:
1. Assume we write a lower version(version = 1) on Server A
2. Server write and crash, topology changed, version = 2.
3. Restart, and load with version = 1. The file is ok, then we call `SetClusterNodes` with `force = false`. So, return status not ok. So `Start` will failed

Reply:
@git-hulk Let's dive into the code:
1. Assume we write a lower version(version = 1) on Server A
2. Server write and crash, topology changed, version = 2.
3. Restart, and load with version = 1. The file is ok, then we call `SetClusterNodes` with `force = false`. So, return status not ok. So `Start` will failed

Reply:
So, maybe we should ignore error if `Load` is failed?

Reply:
The load action will run before the server start, so it won't have any cluster version from the time point. The external controller only can re-sync the new topology after the server was started.

Reply:
Okay, thanks for your patient explaination. But I still think we should ignore it's error:

```
      case LoadState::NodesInfo:
        // std::getline swallowed the newline in the node info, so need to add it back here
        nodesInfo.append(line + "\n");
        break;
```

I think write to os is not atomic, so, we may got a "teared" file if we don't have any checksum. And, `Load` will failed if parsing failed.

Reply:
Yes, that's a good point. I think we can simply write in memory and flush them at once.

Reply:
@git-hulk unless you are writing whole block to the device, you cannot make sure that multiple blocks are atomic sync to OS. Think about the cases below:

1. Write to OS: BLK1 ... BLKn
2. BLK1 - BLKn-1 is sync to disk. But BLKn not
3. System crashed

So, the most convenient way is using RocksDB to store it, or write a checksum.

Reply:
Using OS `rename` API is also possible here. Maybe we can write a atomic-write using `rename`

Reply:
@mapleFU oh, I forgot that it's written to the temp file and renamed after the finish. So it has not this issue.

Reply:
Seems ok. It was my fault, `ofstream::close` will flush the buffer and close the fd. So it's safe to rename it.

What will happen if a crashed server thinks itself is a master node? Will it harm the cluster?